iib/workers/tasks/build_containerized_fbc_operations.py

```diff
@@ -0,0 +1,330 @@
+        with set_registry_token(overwrite_from_index_token, fbc_fragment, append=True):
+            resolved_fbc_fragment = get_resolved_image(fbc_fragment)
+            resolved_fbc_fragments.append(resolved_fbc_fragment)
+
+    prebuild_info = prepare_request_for_build(
+        request_id,
+        RequestConfigFBCOperation(
+            _binary_image=binary_image,
+            from_index=from_index,
+            overwrite_from_index_token=overwrite_from_index_token,
+            add_arches=add_arches,
+            fbc_fragments=fbc_fragments,
+            distribution_scope=distribution_scope,
+            binary_image_config=binary_image_config,
+        ),
+    )
+
+    from_index_resolved = prebuild_info['from_index_resolved']
+    binary_image_resolved = prebuild_info['binary_image_resolved']
+    arches = prebuild_info['arches']
+    # Variable mr_details needs to be assigned, otherwise
+    # cleanup_on_failure() fails when an exception is raised.
+    mr_details: Optional[Dict[str, str]] = None
+
+    with set_registry_token(overwrite_from_index_token, fbc_fragment, append=True):
+        Opm.set_opm_version(from_index_resolved)
+
+    # Store all resolved fragments
+    prebuild_info['fbc_fragments_resolved'] = resolved_fbc_fragments
+
+    # For backward compatibility, only populate old fields if original request used fbc_fragment
+    # This flag should be passed from the API layer
+    if used_fbc_fragment and resolved_fbc_fragments:
+        prebuild_info['fbc_fragment_resolved'] = resolved_fbc_fragments[0]
+
+    _update_index_image_build_state(request_id, prebuild_info)
+
+    with tempfile.TemporaryDirectory(prefix=f'iib-{request_id}-') as temp_dir:
+        artifact_dir = pull_index_db_artifact(
+            from_index,
+            temp_dir,
+        )
+        artifact_index_db_file = os.path.join(artifact_dir, "index.db")
+
+        log.debug("Artifact DB path %s", artifact_index_db_file)
+        if not os.path.exists(artifact_index_db_file):
+            log.error("Artifact DB file not found at %s", artifact_index_db_file)
+            raise IIBError(f"Artifact DB file not found at {artifact_index_db_file}")
+
+        index_git_repo = resolve_git_url(
+            from_index=from_index, index_repo_map=index_to_gitlab_push_map
+        )
+        if not index_git_repo:
+            raise IIBError(f"Cannot resolve the git repository for {from_index}")
+        log.info(
+            "Git repo for %s: %s",
+            from_index,
+            index_git_repo,
+        )
+
+        token_name, git_token = get_git_token(index_git_repo)
+        branch = prebuild_info['ocp_version']
+
+        set_request_state(request_id, 'in_progress', 'Cloning Git repository')
+        local_git_repo_path = os.path.join(temp_dir, 'git', branch)
+        os.makedirs(local_git_repo_path, exist_ok=True)
+
+        # TODO - GitClone takes time
+        #  can we keep the copy and just pull the difference? (6min on dev-env)
```

Comment:
@lipoja let's remove this comment? Now that we know git clone is only pulling one branch ?